.github/workflows/CD.yml

```
@@ -517,50 +495,57 @@ jobs:
       #but using the tools in workflows from the branch triggered with workflow_dispatch.
       - uses: actions/checkout@v2
 
-      - name: Restore cached packages
+      - name: Cache if success
+        id: release-latest-image
         uses: actions/cache@v2
         with:
-          key: "${{ env.PACKAGE_CACHE_KEY }}"
-          path: build/packages
-
-      - name: Load Image
-        run: docker load < build/packages/$IMAGE_NAME.tar
-
-      - name: Configure AWS Credentials
-        uses: aws-actions/configure-aws-credentials@v1
-        with:
-          aws-access-key-id: ${{ secrets.RELEASE_KEY_ID }}
-          aws-secret-access-key: ${{ secrets.RELEASE_SECRET }}
-          aws-region: us-west-2
+          path: |
+            VERSION
+          key: release-latest-image-${{ github.run_id }}
 
       - name: Compare version with Dockerhub latest
         id: version
+        if: steps.release-latest-image.outputs.cache-hit != 'true'
         run: |
-          TAG=`cat build/packages/VERSION`
+          TAG="${{ github.event.inputs.version }}"
```

```suggestion
          TAG="${{ needs.release-checking.outputs.version }}"
```
Don't use the input version even though it should be the same. That goes for all of the other ones as well. It's better to use variables that we've defined.

---

Reasonable. Fixed it with the newest commit. Thought we can reduce the needs step and inputs step if we used the ```${{github.event.inputs.version}}```